### PR TITLE
Feature/at 7147 phone number input

### PR DESCRIPTION
**Pull request by DGabriel-CCPO**
The international phone number component exists in ServiceNow with correct formatting and
Allows the user to select one of the 30 NATO countries at a time, which automatically populates the selected country’s country code.
Selecting ‘United States or ‘Defense Switched Network’ should display correct masking. All other country selections should be generic (no masking implemented).
The international phone number component is added to the following screens in their correct locations:
[‘Let’s confirm your contact information] screen (1.3.1 in Figma)
[‘Let's gather info about your Contracting Officer’s Representative (COR)']screen (1.3.2 in Figma)
[‘Let’s gather info about your ACOR' ]screen (1.3.4 in Figma)
The international phone numbers component is to look as it does in the provided screencap.

### CODE REVIEW

File: src/sass/components/ATATPhoneInput.scss [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+  .v-menu__content.theme--light.menuable__content__active {
+    padding-top: 0px;
+    min-width: 416px !important;
+    width: 420px;
+
+    ._dropdown-text-field + ._suggested {
+      .v-list-item__content {
+        overflow: visible;
+        //margin-top: 26px;
```
Comment (ericyoungquist-ccpo) [inline src/sass/components/ATATPhoneInput.scss:49]: remove commented code

File: src/sass/components/ATATPhoneInput.scss [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            i.v-icon {
+              transform: none !important;
+              color: $primary !important;
+            }
+          }
+        }
+
+      }
+
+      :after, :before {
```
Comment (ericyoungquist-ccpo) [inline src/sass/components/ATATPhoneInput.scss:115]: these probably need `&` ... so `&:after, &:before` -- without the `&` it doesn't attach itself to anything

File: src/sass/utils/flag-icons/_flag-icons-base.scss [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  }
+  &.fis {
+    width: 1em;
+  }
+}
+
+@mixin flag-icon($country) {
+  .fi-#{$country} {
+    background-image: url(#{$flag-icons-path}#{$flag-icons-rect-path}/#{$country}.svg);
+    //&.fis {
```
Comment (ericyoungquist-ccpo) [inline src/sass/utils/flag-icons/_flag-icons-base.scss:24]: remove commented code block

File: src/steps/AcquisitionPackageDetails/COR_ACOR/ContactInfoForm.vue [hunk 4]
```diff
@@ -123,13 +123,15 @@ import {
   AutoCompleteItem, 
   AutoCompleteItemGroups 
 } from "../../../../types/Global";
+import ATATPhoneInput from "@/components/ATATPhoneInput.vue";
```
Comment (ericyoungquist-ccpo) [inline src/steps/AcquisitionPackageDetails/COR_ACOR/ContactInfoForm.vue:126]: move this up alphabetically with the other component imports

File: src/steps/AcquisitionPackageDetails/COR_ACOR/ContactInfoForm.vue [hunk 5]
```diff
@@ -123,13 +123,15 @@ import {
...
 
 @Component({
   components: {
     ATATAutoComplete,
     ATATRadioGroup,
     ATATSelect,
     ATATTextField,
+    ATATPhoneInput
```
Comment (ericyoungquist-ccpo) [inline src/steps/AcquisitionPackageDetails/COR_ACOR/ContactInfoForm.vue:134]: move up alphabetically

File: src/steps/AcquisitionPackageDetails/ContactInfo.vue [hunk 6]
```diff
@@ -122,13 +122,15 @@ import {
   RadioButton, 
   SelectData 
 } from "../../../types/Global";
+import ATATPhoneInput from "@/components/ATATPhoneInput.vue";
```
Comment (ericyoungquist-ccpo) [inline src/steps/AcquisitionPackageDetails/ContactInfo.vue:125]: move up alphabetically with component imports

File: src/steps/AcquisitionPackageDetails/ContactInfo.vue [hunk 7]
```diff
@@ -122,13 +122,15 @@ import {
...
 
 @Component({
   components: {
     ATATAutoComplete,
     ATATTextField,
     ATATSelect,
     ATATRadioGroup,
+    ATATPhoneInput
```
Comment (ericyoungquist-ccpo) [inline src/steps/AcquisitionPackageDetails/ContactInfo.vue:133]: move alphabetically in list

File: types/Global.ts [hunk 8]
```diff
@@ -1,92 +1,99 @@
...
   value: string;
   description?: string;
+}
+
+export interface CountryObj {
+  name: string;
+  countryCode: string;
+  abbreviation: string;
+  suggested?: boolean;
 }
```
Comment (ericyoungquist-ccpo) [inline types/Global.ts:99]: add blank line at end of file

**Review by ericyoungquist-ccpo — CHANGES_REQUESTED** [review]

File: src/sass/components/ATATPhoneInput.scss (hunk 2 above)
Comment (DGabriel-CCPO) [reply src/sass/components/ATATPhoneInput.scss:115]: Screen Shot 2022-03-29 at 9 31 52 AM
adding the & in front of them adds this border under the text input

File: src/components/ATATPhoneInput.vue [hunk 9]
```diff
@@ -0,0 +1,351 @@
...
+            placeholder="Search"
+            @input="searchCountries"
+            append-icon="search"
+            id="DropdownTextField"
+          />
+
+        </template>
+        <template v-slot:item="{ item, on }">
+          <v-list-item class="_country-list"
+                       :class="[item.suggested ? '_suggested' : '', item.active ? '_active' : '']" v-on="on">
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATPhoneInput.vue:41]: fix indentation - see the component below for example

File: src/components/ATATPhoneInput.vue [hunk 10]
```diff
@@ -0,0 +1,351 @@
...
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import {Component, Prop, PropSync} from "vue-property-decorator";
+import ATATAutoComplete from "@/components/ATATAutoComplete.vue";
+import ATATTextField from "@/components/ATATTextField.vue";
+import Inputmask from "inputmask/";
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATPhoneInput.vue:81]: still getting Vetur squiggles on this, and `Instance` lint message about defined but never used
image

**Review by ericyoungquist-ccpo — CHANGES_REQUESTED** [review]
just a couple minor things and should be good!

**Review by ericyoungquist-ccpo — APPROVED** [review]

**Pull request merged**